plugins/woocommerce/src/StoreApi/Authentication.php

```diff
@@ -22,6 +23,21 @@ public function init() {
 		add_filter( 'rest_pre_serve_request', array( $this, 'send_cors_headers' ), 10, 3 );
 		add_filter( 'rest_allowed_cors_headers', array( $this, 'allowed_cors_headers' ) );
 
+		// If cart has a valid token, override the core session class.
+		if ( isset( $_SERVER['HTTP_CART_TOKEN'] ) ) {
+			$cart_token      = wc_clean( wp_unslash( $_SERVER['HTTP_CART_TOKEN'] ?? '' ) );
+			$has_valid_token = JsonWebToken::validate( $cart_token, $this->get_cart_token_secret() );
+			if ( $has_valid_token ) {
+				// Overrides the core session class.
+				add_filter(
+					'woocommerce_session_handler',
+					function () {
+						return SessionHandler::class;
+					}
+				);
+			}
+		}
```

Comment:
Since we have a get_cart_token method here now we can make this cleaner:

```suggestion
        // Validate returns early if no token, so no perf penalty
		if ( JsonWebToken::validate( $this->get_cart_token(), $this->get_cart_token_secret() ) ) {
			// Overrides the core session class.
			add_filter(
				'woocommerce_session_handler',
				function () {
					return SessionHandler::class;
				}
			);
		}
```

Reply:
https://github.com/woocommerce/woocommerce/pull/55047/commits/84cf630021cd8724c4b437e29243433d8d95f364